### PR TITLE
Ensure that GeneratorReturnExceptions are not handled by InterpreterStackFrame::DebugProcess

### PR DESCRIPTION
**Pull request by paolosevMSFT**
When running a generator function with the -debuglaunch flag the engine crashes with simple code like:
```
  function* test() {
    try {
      yield 42;
    } catch (e) {
    }
  }
  var t = test();
  t.next();
  var v = t.return();
```

The reason is that a special exception is thrown by JavascriptOperators::OP_ResumeYield() to carry the return value from a generator. When the generator has a try/catch, like in this case, this exception is caught and rethrown by InterpreterStackFrame::OP_TryCatch(), and after that it is caught and handled by JavascriptGenerator::CallGenerator().

But when we run with -debuglaunch the exception gets caught by InterpreterStackFrame::DebugProcess(), which does not rethrow it. Therefore the generator code does not have a chance to complete correctly and the process crashes because an invalid var is returned in place of a generator yielded object.

As fix this PR adds a check for IsGeneratorReturnException() in InterpreterStackFrame::DebugProcess() to avoid skipping this exception.

### CODE REVIEW

File: test/es6/rlexe.xml [hunk 1]
```diff
@@ -868,6 +868,13 @@
       <tags>exclude_arm</tags>
     </default>
   </test>
+  <test>
+    <default>
+      <files>generators-functionality.js</files>
+      <compile-flags>-ES6Generators -ES6Classes -ES6DefaultArgs -debuglaunch -args summary -endargs</compile-flags>
```
Comment (paolosevMSFT) [inline test/es6/rlexe.xml]: Note that a lot of existing generators-functionality.js tests were already failing/crashing if we added the -debuglaunch flag.

Comment (akroshg) [reply test/es6/rlexe.xml]: interesting.

---
In reply to: [201881156](https://github.com/Microsoft/ChakraCore/pull/5451#discussion_r201881156) [](ancestors = 201881156)

Comment (akroshg) [reply test/es6/rlexe.xml]: Could you  add with async functionality as well?

---
In reply to: [201881720](https://github.com/Microsoft/ChakraCore/pull/5451#discussion_r201881720) [](ancestors = 201881720,201881156)

**Comment by akroshg** [comment]
Looks good to me. Aneesh can you chime in if more tests needed to be added?

**Comment by meg-gupta** [comment]
LGTM

File: test/es6/rlexe.xml (hunk 1 above)
Comment (aneeshdk) [reply test/es6/rlexe.xml]: Do we already have test case which has the failing scenario in the bug? If not can you please add it?

Comment (aneeshdk) [reply test/es6/rlexe.xml]: Can you please add a test case with the same scenario where we catch first chance exception? Also a test case with finally block?

Comment (paolosevMSFT) [reply test/es6/rlexe.xml]: There is already a test case with the same scenario: 
```
        name: "Test cases with return api invoked while the generator resumes inside one of the try-catch-finally block",
        body: function () {
            var gf = function* () { try { yield 1; } catch (ex) { }}
            var g = gf();
            assert.areEqual({ value: 1, done: false }, g.next(), "First next call should return 1 from try block");
            assert.areEqual({ value: 2, done: true }, g.return(2), "Return statement completes the function from try block");
            assert.areEqual({ value: undefined, done: true }, g.next(), "Generator is already in completed state");
```

Comment (paolosevMSFT) [reply test/es6/rlexe.xml]: There is also an existing test with same scenario + finally block:

```
            gf = function* () { try { throw 1; } catch (e) { yield e; } finally { yield 2; return 3; } }
            g = gf();
            assert.areEqual({ value: 1, done: false }, g.next(), "First next call should return 1 from the catch block");
            assert.areEqual({ value: 2, done: false }, g.return(100), "Return causes the finally block to execute");
            assert.areEqual({ value: 3, done: true }, g.next(), "Second next call should return 3 from the end of finally block");
            assert.areEqual({ value: undefined, done: true }, g.next(), "Method execution has finished");
```

**Comment by aneeshdk** [comment]
LGTM

**Comment by akroshg** [comment]
Seems fine to me.

**Pull request merged**